### PR TITLE
Make BASEPKGS optionally less 'helpful'

### LINKED ISSUE #35: [Feature Request] Population of BASEPKGS var should be conditional
**Describe the bug**

The [`BASEPKGS`](https://github.com/plus3it/amigen8/blob/master/OSpackages.sh#L164) array-setup within the `OSpackages.sh` script's [`PrepChroot`](https://github.com/plus3it/amigen8/blob/master/OSpackages.sh#L160) function should only reference the [`/etc/os-release`](https://github.com/plus3it/amigen8/blob/master/OSpackages.sh#L165) and [`/etc/yum.repos.d/*`](https://github.com/plus3it/amigen8/blob/master/OSpackages.sh#L166) files when doing a same-platform AMI-creation

**Severity**

Requires manual removal of offending lines when doing cross-distro bootstrapper-builds (see #34)

**Expected behavior**

When using the scripts to do a cross-distro bootstrapper-build, release and repository content should come _only_ from the RPMs specified via the `-r`/`--repo-rpms` arguments.

**Deviance Description**

Regardless of use-case, script attempts to suck in RPMs by referencing the  `/etc/os-release` and `/etc/yum.repos.d/*` files

**Screenshots**

**Additional context**

**Fix Suggestions**

Add in a switch or other logic to disable the offending behavior

**References**

Feature Request #34

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #35

### CODE REVIEW

File: OSpackages.sh [hunk 1]
```diff
@@ -163,12 +165,19 @@ function PrepChroot {
...
-   mapfile -t BASEPKGS < <(
-      rpm --qf '%{name}\n' -qf /etc/os-release ; \
-      rpm --qf '%{name}\n' -qf  /etc/yum.repos.d/* 2>&1 | grep -v "not owned" | sort -u ; \
-      echo yum-utils
+   BASEPKGS=(
+      yum-utils
    )
 
+   # Don't try to be helpful if doing cross-distro (i.e., "bootstrapper-build")
+   if [[ -z ${ISCROSSDISTRO} ]]
```
Comment (lorengordon) [inline OSpackages.sh:173]: Does this need a default assignment to support `set -u`?

```suggestion
   if [[ -z ${ISCROSSDISTRO:-} ]]
```

Comment (ferricoxide) [reply OSpackages.sh:173]: I ran it past `shellcheck` and it didn't bark, but, probably run into if were to enable full strict-mode?

Comment (lorengordon) [reply OSpackages.sh]: Yeah, I think by default shellcheck evaluates strictly only if the related options are set in the shebang or at the top of the file.

File: OSpackages.sh [hunk 2]
```diff
@@ -415,6 +424,10 @@ do
                   ;;
             esac
             ;;
+      -X|--cross-distro)
+            ISCROSSDISTRO=TRUE
+            esac
```
Comment (lorengordon) [inline OSpackages.sh:429]: This `esac` seems like it may be extraneous here?

Comment (ferricoxide) [reply OSpackages.sh]: Heh... came back from pushing the fix to finding this. Any way, yes, happened when I was resolving conflicts in the damned GUI. 😛

**Comment by lorengordon** [comment]
Building this branch: https://console.aws.amazon.com/codesuite/codebuild/701759196663/projects/spel-ci/build/spel-ci%3A56e4e4ad-e912-4b99-a7b2-97f8285a0a39?region=us-east-1

**Review by lorengordon — APPROVED** [review]
build is passing! 🚀

**Pull request merged**